Build/Xamarin.Build/XamPub/MSBuild/GeneratePublicationManifest.cs

```diff
@@ -119,10 +122,15 @@ TFile ProcessFileBase<TFile> (string path, string evergreenName = null)
                 Branch = gitBranch
             };
 
-            releaseFile.PublishUri = $"{BasePublishUri}/{fileName}";
+            // deal with MSBuild converting / to \, multiple //, etc.
+            string MakePublishUri (string relativePath)
+                => Regex.Replace ($"{BasePublishUri}/{relativePath}".Replace ("\\", "/"), @"\/+", "/")
+                    .Replace ("https:/", "https://");
+
+            releaseFile.PublishUri = MakePublishUri (fileName);
 
             if (!string.IsNullOrEmpty (evergreenName))
-                releaseFile.EvergreenUri = $"{BasePublishUri}/{evergreenName}";
+                releaseFile.EvergreenUri = MakePublishUri (evergreenName);
```

Comment:
The issue this attempts to fix isn't actually happening here, it's happening in the _other_ set of `releaseFile.EvergreenUri`. The code looks like this:

```csharp
releaseFile.EvergreenUri =
                    Path.GetDirectoryName (relativePath) + "/" +
                    updaterItem.Groups ["name"].Value +
                    fileExtension;
```
and should look like this:
```csharp
releaseFile.EvergreenUri =
                    relativePath.Replace (Path.GetFileName (relativePath), string.Empty) +
                    updaterItem.Groups ["name"].Value +
                    fileExtension;
```

Path.GetDirectoryName mangles URIs, unsurprisingly.

Reply:
In my testing, you don't need this code at all, with the ^ fix applied.

Reply:
This could also be written using `System.Uri` and the `Segments` property--parse `relativePath` into a URI, remove the last segment, and then graft the evergreen filename on, but that's neither here nor there.